stores/sql/mysql/main.go

```diff
@@ -566,6 +570,63 @@ func (tx *MainDatabaseTx) ProcessChainUpdate(ctx context.Context, fn func(ssql.C
 	})
 }
 
+func (tx *MainDatabaseTx) PrunableContractRoots(ctx context.Context, fcid types.FileContractID, roots []types.Hash256) (indices []uint64, err error) {
+	// build tmp table name
+	var r [4]byte
+	frand.Read(r[:])
+	tmpTable := strings.ReplaceAll(fmt.Sprintf("diff_%s_%x", fcid.String()[:8], r), ":", "_")
+
+	// create temporary table
+	_, err = tx.Exec(ctx, fmt.Sprintf(`
+CREATE TEMPORARY TABLE %s (idx INT, root blob);
+CREATE INDEX %s_idx ON %s (root(32));`, tmpTable, tmpTable, tmpTable))
```

Comment:
I had a couple of options here, the reason we have to duplicate `PrunableContractRoots` is because of this `INDEX`. In MySQL it requires a length `BLOB/TEXT column 'root' used in key specification without a key length`. I considered splitting the query into creating the temporary table and then fetching roots, that way we can move the part starting from "execute query" to the main file. Another option is to add something like `withTemporryTable(ctx, setupQuery, teardownQuery, fn (tmpTableName) error)`. Wasn't sure of the reusability of something like that so decided to just duplicate the code for now... wdyt?